src/main/golem/util/validation/README.md

```diff
@@ -0,0 +1,102 @@
+# Matrix Validation DSL
+
+This package provides definitions for a domain-specific language that you can use to validate matrices used as inputs to your functions in a consistent way. The rules throw detailed, human-readable exceptions using an appropriate Java exception class based on rules that look like this:
+
+```kotlin
```

Comment:
put the "import XXX" line at the top of the code snippet here to let people know what they'd have to import to actually run this.